thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/ThirdEyeAnomalyApplication.java

```diff
@@ -73,13 +85,14 @@ public void start() throws Exception {
 
         if (config.isWorker()) {
           anomalyFunctionFactory = new AnomalyFunctionFactory(config.getFunctionConfigPath());
+
           taskDriver = new TaskDriver(config, anomalyFunctionFactory);
           taskDriver.start();
         }
         if (config.isScheduler()) {
           detectionJobScheduler = new DetectionJobScheduler();
           detectionJobScheduler.start();
-          environment.jersey().register(new DetectionJobResource(detectionJobScheduler));
+          environment.jersey().register(new DetectionJobResource(detectionJobScheduler, config.getFilterAutotuneConfigPath(), config.getAlertFilterConfigPath()));
```

Comment:
Pass in AlertFilterAutotuneFactory and AlertFilterFactory objects instead of their config path, so we could share the factories across different Resources.